docs/lti-integration.md

```diff
@@ -0,0 +1,159 @@
+integration in 2024.
+
+The following documentation details the steps required to install Code.org as an
+LTI tool in Canvas. As an outcome of following these steps, you will supply the
+generated `Client ID` and the `Deployment ID` to Code.org as the final
+configuration step.
+
+## References
+
+Canvas provides installation and configuration instructions for LTI Tools. Use
+these instructions, and refer to the steps below for Code.org specific details.
+
+- [Create a new LTI Key][install-canvas] which configures Code.org as an LTI Tool
+- [Configure the LTI app][configure-canvas] to generate the Deployment ID
+
+[install-canvas]: https://community.canvaslms.com/t5/Admin-Guide/How-do-I-configure-an-LTI-key-for-an-account/ta-p/140
+[configure-canvas]: https://community.canvaslms.com/t5/Admin-Guide/How-do-I-configure-an-external-app-for-an-account-using-a-client/ta-p/202
+
+## Installing Code.org in Canvas
+
+You will follow [these instructions][install-canvas] for configuring an LTI Key,
+i.e. installing an LTI Tool in Canavs. The outcome of this installation will be
+a Client ID to provide Code.org (in [Step 4](#step-4))
+
+### Step 1
+
+1. From the Admin menu, select `Developer Keys`
+1. Click `Add Developer Key` button, and select the `LTI Key` option
+1. Enter LTI Key Settings:
+    - Key Name: `<your-key-name>`
+    - Owner Email: `<admin-owner-email>`
+    - Redirect URIs: `https://studio.code.org`
+    - Notes: `Optional: Any notes about the LTI key, such as the reason it was created.`
+
+### Step 2
+
+There are two options for entering configuration settings. The easiest is using
+JSON. Optionally, you can chose to manually enter these settings.
+
+#### Enter JSON Details
+
+The easiest way to configure the LTI Tool in Canvas is via the `Paste JSON`
+configuration. Below is the JSON you can paste directly into the
+`LTI 1.3 Configuration` field:
+
+1. Select the `Paste JSON` option from the Configuration dropdown menu
+1. Past the below JSON into the field
+1. Click `Save`
+
+```json
+{
+  "title": "Code.org",
+  "description": "Code.org LTI Tool",
+  "oidc_initiation_url": "https://studio.code.org/lti/v1/login",
+  "target_link_uri": "https://studio.code.org/home",
+  "scopes": [
+    "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/lineitem.readonly",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly",
+    "https://purl.imsglobal.org/spec/lti-ags/scope/score",
+    "https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly",
+    "https://canvas.instructure.com/lti/public_jwk/scope/update",
+    "https://canvas.instructure.com/lti/account_lookup/scope/show",
+    "https://canvas.instructure.com/lti-ags/progress/scope/show"
+  ],
+  "extensions": [
+    {
+      "tool_id": "code-dot-org",
+      "platform": "canvas.instructure.com",
+      "privacy_level": "public",
+      "settings": {
+        "text": "Launch Code.org",
```

Comment:
`text` is what will show up as the hyperlink when adding the LTI tool in various placements. Alternatively, you can override that value with a `text` key in each of the placements. This allows us to point them to particular URLs depending on where they are accessing the LTI tool. For example, when we want to surface deep linking interface, we can specify that in the particular placement.